src/main/java/duke/command/DeleteCommand.java

```diff
@@ -0,0 +1,52 @@
+package duke.command;
+
+import duke.task.Task;
+import duke.util.DukeException;
+import duke.util.Storage;
+import duke.util.TaskList;
+import duke.util.Ui;
+
+/**
+ * Represents a delete command to delete a task from the list.
+ */
+public class DeleteCommand extends Command {
+    private final int taskNum;
+
+    /**
+     * Class constructor.
+     *
+     * @param taskNum The index of the task to be deleted.
+     */
+    public DeleteCommand(int taskNum) {
+        this.taskNum = taskNum;
+    }
+
+    /**
+     * Executes the delete command, prints confirmation message and saves the updated task list to file.
+     *
+     * @param taskList The current list of tasks.
+     * @param ui The ui of the program.
+     * @param storage The storage of the program.
+     * @throws DukeException If updated task list cannot be saved to file, or if the task number given is invalid.
+     */
+    public void execute(TaskList taskList, Ui ui, Storage storage) throws DukeException {
+        int currentTotalTasks = taskList.getTotalTasks();
+        if (taskNum <= currentTotalTasks && taskNum > 0) {
+            Task taskDeleted = taskList.delete(this.taskNum);
+            ui.printOutput("Okay, I've deleted this task:\n  " + ui.showIndent()
+                    + taskDeleted + "\n" + ui.showIndent() + taskList.getListStatus());
+            storage.saveToHardDisk(taskList);
+        } else {
+            throw new DukeException("Invalid task number. There are " + currentTotalTasks + " in the list.");
```

Comment:
I like how detailed your `DukeException` messages are. I can definitely see it helping the user.